app/models/cinstance.rb

```diff
@@ -147,8 +147,9 @@ def validate_plan_is_unique?
     where(['cinstances.created_at <= ?', period.end])
   }
 
-  def self.provided_by(account)
-    joins(:service).references(:service).merge(Service.of_account(account)).readonly(false)
+  def self.provided_by(account, service_filter: nil)
+    service_filter ||= -> { _1 }
+    where(plan_id: ApplicationPlan.unscoped.where(issuer_type: "Service", issuer_id: service_filter[Service.unscoped.select(:id).of_account(account)]))
```

Comment:
This is a little tricky. To avoid redundant sub-queries from  `Admin::Api::ApplicationsController#applications` I wanted to be able to merge additional conditions to the service sub-query here. I couldn't figure out ohw to do it better. If you have good ideas, let me know.

Reply:
I've got rid of it.